src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -94,8 +94,19 @@ protected function forward(string $controller, array $path = array(), array $que
      *
      * @final
      */
-    protected function redirect(string $url, int $status = 302): RedirectResponse
+    protected function redirect(string $url, int $status = 302, bool $keepQueryParams = false): RedirectResponse
```

Comment:
As this function accepts a URL and not a route, I don't see the usefulness of this feature to be honest. When generating the URL, you can already use the url generator to add the current query string based on `$request->query->all();`.

Reply:
redirect() is called by redirectToRoute() and I wanted to keep things consistent with both functions.

Reply:
if you call `redirectToRoute`, you can pass extra params to the url generator, and it will put them in the query string.

So IMO, this is not needed (the redirect controller does not give you the control of the arguments which is why it needed to add this feature)